bundles/org.eclipse.core.resources/src/org/eclipse/core/internal/resources/ProjectPreferences.java

```diff
@@ -756,4 +759,16 @@ private void silentLoad() {
 			node.setLoading(false);
 		}
 	}
+
+	public void setWorkspace(Workspace workspace) {
+		this.workspace = workspace;
+	}
+
+	private Workspace getWorkspace() {
+		if (workspace != null) {
+			return workspace;
+		}
+		// last resort...
```

Comment:
The code in `Workspace` below inits this to non-null value, but only **after** it is already initialized here by using "last resort". I understand that breaking this here could be problematic for someone who always created ProjectPreferences via default public constructor and assumed that workspace is "magically known", but this seem to be not clean. I have however no better proposal.

Reply:
The problem is described here but I think we won't have a fast fix:

- https://github.com/eclipse-equinox/equinox.bundles/issues/45

so I hope we find a way for an intermediate solution.